runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -579,72 +578,91 @@ public void start() {
     sampler.start();
 
     // Periodically report workers counters and other updates.
-    globalWorkerUpdatesTimer = executorSupplier.apply("GlobalWorkerUpdatesTimer");
-    globalWorkerUpdatesTimer.scheduleWithFixedDelay(
-        this::reportPeriodicWorkerUpdates,
-        0,
-        options.getWindmillHarnessUpdateReportingPeriod().getMillis(),
-        TimeUnit.MILLISECONDS);
-
-    refreshWorkTimer = executorSupplier.apply("RefreshWork");
+    timersMap.put("GlobalWorkerUpdatesTimer", executorSupplier.apply("GlobalWorkerUpdates"));
+    timersMap
+        .get("GlobalWorkerUpdatesTimer")
+        .scheduleWithFixedDelay(
+            this::reportPeriodicWorkerUpdates,
+            0,
+            options.getWindmillHarnessUpdateReportingPeriod().getMillis(),
+            TimeUnit.MILLISECONDS);
+
+    timersMap.put("WorkerMessageTimer", executorSupplier.apply("ReportWorkerMessage"));
```

Comment:
no need to create this executor if not actually scheduling anything on it, see above.

Reply:
Done.